### PR TITLE
Update CHANGELOG for 0.48.0 release

### PR DESCRIPTION
**Pull request by tigrannajaryan**
I also fixed some PR numbers in existing entries.

Contributes to https://github.com/open-telemetry/opentelemetry-collector/issues/5120

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,16 @@
 
 ### 🛑 Breaking changes 🛑
 
+### 🚩 Deprecations 🚩
```
Comment (jpkrohling) [inline CHANGELOG.md:7]: Nice, we should indeed add those sections during the release.

Comment (jpkrohling) [reply CHANGELOG.md:7]: It is missing the "New components" section though, isn't it?

Comment (mx-psi) [reply CHANGELOG.md:7]: I guess we don't usually have New Components on the core repo? Last time we used that section on the core repo changelog was on v0.14.0

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,24 +26,25 @@
   - "instrumentationLibraryMetrics" is now "scopeMetrics"
   - "instrumentationLibrarySpans" is now "scopeSpans"
   - "instrumentationLibrary" is now "scope"
+- AsString for pdata.AttributeValue now returns the JSON-encoded string of floats. (#4934)
```
Comment (Aneurysm9) [inline CHANGELOG.md:29]:
```suggestion
- AsString for pdata.Value now returns the JSON-encoded string of floats. (#4934)
```

Should we use the new name or the old name?  Or both? Do they both do this now?

**Review by bogdandrutu — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (tigrannajaryan) [reply CHANGELOG.md:7]: I copied the sections that existed and deleted the content. I don't think we have new components in this release and there are no immediate plans to add new components to this repo, so probably don't need that section anyway.

**Review by dmitryax — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Pull request merged**